instaloader/instaloadercontext.py

```diff
@@ -351,7 +351,7 @@ def get_json(self, path: str, params: Dict[str, Any], host: str = 'www.instagram
                     raise ConnectionException("\"window._sharedData\" does not contain required keys.")
                 # If GraphQL data is missing in `window._sharedData`, search for it in `__additionalDataLoaded`.
                 if 'graphql' not in post_or_profile_page[0]:
-                    match = re.search(r'window\.__additionalDataLoaded\([^{]+{"graphql":({.*})}\);</script>',
+                    match = re.search(r'window\.__additionalDataLoaded\([^{]+.*"graphql":({.*}).*}\);<\/script>',
                                       resp.text)
                     if match is not None:
                         post_or_profile_page[0]['graphql'] = json.loads(match.group(1))
```

Comment:
```suggestion
                    match = re.search(r'window\.__additionalDataLoaded\(.*({"graphql":.*})\);</script>',
                                      resp.text)
                    if match is not None:
                        post_or_profile_page[0]['graphql'] = json.loads(match.group(1))['graphql']
```

Reply:
@Thammus Thanks your comment. But, `({"` seems not correct. `"graphql"` doesn't appear at first. So, should be `({.*"`?

Reply:
Ah, okay. You are right. If `"graphql"` is not the first key in the object, additional adjustments to the regex are necessary in order to match correctly. Simply adjusting `({.*"` as you suggest could cause other problems though. Consider the regular expression 
```regexp
window\.__additionalDataLoaded\(.*({.*"graphql":.*})\);</script>
```
and the test string
```javascript
window.__additionalDataLoaded(path, {"first_key": {"something": 123}, "graphql": {"foo": "bar", "baz": {"foo": "bar"}}, "something_different": {"foo": "baz"}});</script>
```
then just the string
```json
{"something": 123}, "graphql": {"foo": "bar", "baz": {"foo": "bar"}}, "something_different": {"foo": "baz"}}
```
is matched, which is not syntactically correct JSON. However, this can be corrected by making the very first `.*` in the regex "lazy" i.e. "non-greedy" by using `.*?` instead.

Reply:
```suggestion
                    match = re.search(r'window\.__additionalDataLoaded\(.*?({.*"graphql":.*})\);</script>',
                                      resp.text)
                    if match is not None:
                        post_or_profile_page[0]['graphql'] = json.loads(match.group(1))['graphql']
```

Reply:
@Thammus ok. thanks👍  I fixed.

Reply:
If you are using `json.loads` just access the `json.loads(match.group('json'))['graphql']` field from the parsed json object and use a simple and straightforward regex `r'window\.__additionalDataLoaded\(.*?,(?P<json>.*)\);</script>'` to extract the json data. There is no need to parse json data and extract its field in a complex regex.